shared/js/initApp.js

```diff
@@ -150,7 +150,35 @@ $(window).on('function_editor_closed', function() {
   }
 })(appOptions.level);
 
+dashboard.saveProject = function(callback) {
+  var app_id = dashboard.currentApp.id;
+  dashboard.currentApp.startBlocks = Blockly.Xml.domToText(Blockly.Xml.blockSpaceToDom(Blockly.mainBlockSpace));
+  if (app_id) {
+    storageApps().update(app_id, dashboard.currentApp, function(data) {
+      callback(data);
+    });
+  } else {
+    storageApps().create(dashboard.currentApp, function(data) {
+      if (history) {
+        history.pushState({}, '', '?id=' + data.id);
+      }
+      callback(data);
+    });
+  }
+};
+
 function initApp() {
+  if (appOptions.level.isProject) {
+    if (dashboard.currentApp) {
+      appOptions.level.startBlocks = dashboard.currentApp.startBlocks;
```

Comment:
maybe we should name this something other than startBlocks. source (as in source code) would kind of match the existing usage (LevelSource), but I'm not in love with that name because it's kind of ambiguous with source as in source/destination. Also it's might not always be "blocks".